### DISCUSSION TITLE
GradCAM/GradCAMpp for ViT

### DISCUSSION THREAD

**Discussion by a-parida12** [Project-MONAI/MONAI#6900] [Q&A]
Is there a tutorial or a way to vizualize the GradCAM saliency maps for the ViT model. 
I use the code below-
```python
from monai.visualize.class_activation_maps import GradCAM
# classifier.model:ViT()
target_layers = [classifier.model.blocks[-1].norm1]
grad_cam = GradCAM(classifier.model.to(device), target_layers=target_layers)

samples = samples.to(device, non_blocking=True)
maps = grad_cam.compute_map(samples, class_idx=targets)

**Comment by KumoLiu** [comment]
Hi @a-parida12, We don't have a specific tutorial on how to get GradCAM for ViT models, but there is a folder on interpretability that you might find useful.
https://github.com/Project-MONAI/tutorials/tree/main/modules/interpretability

I hope it helps, thanks!
